src/DevCenter/DevCenter.AutoRest/custom/New-AzDevCenterAdminCatalog.ps1

```diff
@@ -0,0 +1,240 @@
+
+BODY <ICatalog>: Represents a catalog.
+  [SystemDataCreatedAt <DateTime?>]: The timestamp of resource creation (UTC).
+  [SystemDataCreatedBy <String>]: The identity that created the resource.
+  [SystemDataCreatedByType <CreatedByType?>]: The type of identity that created the resource.
+  [SystemDataLastModifiedAt <DateTime?>]: The timestamp of resource last modification (UTC)
+  [SystemDataLastModifiedBy <String>]: The identity that last modified the resource.
+  [SystemDataLastModifiedByType <CreatedByType?>]: The type of identity that last modified the resource.
+  [AdoGitBranch <String>]: Git branch.
+  [AdoGitPath <String>]: The folder where the catalog items can be found inside the repository.
+  [AdoGitSecretIdentifier <String>]: A reference to the Key Vault secret containing a security token to authenticate to a Git repository.
+  [AdoGitUri <String>]: Git URI.
+  [GitHubBranch <String>]: Git branch.
+  [GitHubPath <String>]: The folder where the catalog items can be found inside the repository.
+  [GitHubSecretIdentifier <String>]: A reference to the Key Vault secret containing a security token to authenticate to a Git repository.
+  [GitHubUri <String>]: Git URI.
+
+INPUTOBJECT <IDevCenterIdentity>: Identity Parameter
+  [AttachedNetworkConnectionName <String>]: The name of the attached NetworkConnection.
+  [CatalogName <String>]: The name of the Catalog.
+  [DevBoxDefinitionName <String>]: The name of the Dev Box definition.
+  [DevCenterName <String>]: The name of the devcenter.
+  [EnvironmentTypeName <String>]: The name of the environment type.
+  [GalleryName <String>]: The name of the gallery.
+  [Id <String>]: Resource identity path
+  [ImageName <String>]: The name of the image.
+  [Location <String>]: The Azure region
+  [NetworkConnectionName <String>]: Name of the Network Connection that can be applied to a Pool.
+  [OperationId <String>]: The ID of an ongoing async operation
+  [PoolName <String>]: Name of the pool.
+  [ProjectName <String>]: The name of the project.
+  [ResourceGroupName <String>]: The name of the resource group. The name is case insensitive.
+  [ScheduleName <String>]: The name of the schedule that uniquely identifies it.
+  [SubscriptionId <String>]: The ID of the target subscription.
+  [VersionName <String>]: The version of the image.
+.Link
+https://learn.microsoft.com/powershell/module/az.devcenter/new-azdevcenteradmincatalog
+#>
+function New-AzDevCenterAdminCatalog {
+[OutputType([Microsoft.Azure.PowerShell.Cmdlets.DevCenter.Models.Api20230401.ICatalog])]
+[CmdletBinding(DefaultParameterSetName='CreateExpandedAdo', PositionalBinding=$false, SupportsShouldProcess, ConfirmImpact='Medium')]
+param(
+    [Parameter(ParameterSetName='CreateExpandedAdo', Mandatory)]
+    [Parameter(ParameterSetName='CreateExpandedGitHub', Mandatory)]
+    [Microsoft.Azure.PowerShell.Cmdlets.DevCenter.Category('Path')]
```

Comment:
Could you explain what's the purpose of this custom code? compared to the one in the internal folder.

Reply:
I added an additional variant to separate the ADO and Github parameters. Is there a way to do this using a directive?